integration/utils/follower.go

```diff
@@ -128,9 +128,20 @@ Loop:
 		getBlockByHeightDuration := time.Since(GetBlockByHeightTime)
 
 		var blockTxs, blockUnknownTxs uint64
+
+		for _, guaranteed := range block.CollectionGuarantees {
+			_, err := f.client.GetCollection(f.ctx, guaranteed.CollectionID)
+			if err != nil {
+				continue Loop // silently try again if not ready; can happen several dozen times
+			}
+		}
+
 		for _, guaranteed := range block.CollectionGuarantees {
 			col, err := f.client.GetCollection(f.ctx, guaranteed.CollectionID)
 			if err != nil {
```

Comment:
This is where the unknown TX stat bug occurred before this PR. How? The entire for loop for the block would start dozen or two dozen times, maybe process a collection, but also get into this error branch and error out. However, because some TXs in some collections did get processed and deleted from the map, this means that if an error out occurs after being deleted from the map, then the next attempt to process the block will find the same TX (already deleted from the map) and wrongly count it as an 'unknown TX'...